code/game/objects/effects/gibs.dm

```diff
@@ -18,36 +18,38 @@
 
 /obj/effect/gibspawner/New(location, datum/dna/MobDNA)
 	..()
+	ASSERT(length(gibtypes) == length(gibamounts))
+	ASSERT(length(gibamounts) == length(gibdirections))
 
-	if(istype(loc,/turf)) //basically if a badmin spawns it
+	if(istype(loc, /turf)) //basically if a badmin spawns it
 		Gib(loc, MobDNA)
 
-/obj/effect/gibspawner/proc/Gib(atom/location, datum/dna/MobDNA = null)
-	if(gibtypes.len != gibamounts.len || gibamounts.len != gibdirections.len)
-		to_chat(world, "<span class='warning'>Gib list length mismatch!</span>")
-		return
-
+/obj/effect/gibspawner/proc/Gib(atom/location, datum/dna/MobDNA)
 	var/obj/effect/decal/cleanable/blood/gibs/gib = null
 
 	if(sparks)
 		do_sparks(2, 1, location)
 
-	for(var/i = 1, i<= gibtypes.len, i++)
-		if(gibamounts[i])
-			for(var/j = 1, j<= gibamounts[i], j++)
-				var/gibType = gibtypes[i]
-				gib = new gibType(location)
-
-
-				gib.blood_DNA = list()
-				if(MobDNA)
-					gib.blood_DNA[MobDNA.unique_enzymes] = MobDNA.blood_type
-				else if(istype(src, /obj/effect/gibspawner/xeno))
-					gib.blood_DNA["UNKNOWN DNA"] = "X*"
-				else if(istype(src, /obj/effect/gibspawner/human)) // Probably a monkey
-					gib.blood_DNA["Non-human DNA"] = "A+"
-				var/list/directions = gibdirections[i]
-				if(directions.len)
-					gib.streak(directions)
-
-	qdel(src)
+	for(var/I in 1 to length(gibtypes))
```

Comment:
Important distinction: `I` should be preferred for items, not numbers -- I would make this lowercase again.

Reply:
I wasn't aware there was a distinction, I've always preferred uppercase `I`s over lowercase even outside of DM coding.